src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -967,17 +967,20 @@ protected int engineDoFinal(final ByteBuffer input, final ByteBuffer output)
     return output.position() - initialPosition;
   }
 
-  private void checkOutputBuffer(final int inputLength, final byte[] output, final int outputOffset)
+  private void checkOutputBuffer(
+      final int inputLength, final byte[] output, final int outputOffset, boolean doFinal)
       throws ShortBufferException {
     if (inputLength < 0 || outputOffset < 0 || outputOffset > output.length) {
```

Comment:
Does `outputOffset == output.length` make sense?

Reply:
Good catch! That's yet another bug in this routine. That case equates to writing to the first byte beyond the array limits.